### PR TITLE
Remove thor + Added a `TranslationObserver` feature which enables the user to handle MissingResource events

### PR DESCRIPTION
**Pull request by gmiserez**
Removes all thor references.

Translations: 
- A `TranslationObserver` class implementation can be be added by the user of the package to handle the logging himself (#22)

Tracking:
- TBD

### CODE REVIEW

**Review by glucaci — CHANGES_REQUESTED** [review]

**Comment by gmiserez** [comment]
Added implementation of Observer class (enhancement #22)

File: src/HotChocolate.Extensions.Tracking/Persistence/TrackingHostedService.cs [hunk 1]
```diff
@@ -15,30 +15,33 @@ public sealed class TrackingHostedService : BackgroundService
...
 
     protected override async Task ExecuteAsync(CancellationToken stoppingToken)
     {
-        Log.TrackingPipelineStarted();
+        _logger.LogInformation("Tracking Pipeline Started");
 
         await foreach (TrackingMessage message in ReadAllAsync(stoppingToken))
         {
-            Log.SavingTrackingMessage(message.TrackingEntry);
+            _logger.LogInformation("Saving tracking message", message.TrackingEntry);
```
[the remark this answers is not in the document]
Comment (gmiserez) [reply src/HotChocolate.Extensions.Tracking/Persistence/TrackingHostedService.cs]: good point.

File: src/HotChocolate.Extensions.Tracking/Persistence/TrackingHostedService.cs [hunk 2]
```diff
@@ -15,30 +15,33 @@ public sealed class TrackingHostedService : BackgroundService
...
         _trackingRepositoryFactory = trackingRepositoryFactory;
+        _logger = logger;
     }
 
     protected override async Task ExecuteAsync(CancellationToken stoppingToken)
     {
-        Log.TrackingPipelineStarted();
+        _logger.LogInformation("Tracking Pipeline Started");
 
         await foreach (TrackingMessage message in ReadAllAsync(stoppingToken))
```
[the remark this answers is not in the document]
Comment (gmiserez) [reply src/HotChocolate.Extensions.Tracking/Persistence/TrackingHostedService.cs]: Good point. Done.

File: src/HotChocolate.Extensions.Tracking/Persistence/TrackingHostedService.cs [hunk 3]
```diff
@@ -15,30 +15,33 @@ public sealed class TrackingHostedService : BackgroundService
...
+    private readonly ILogger<TrackingHostedService> _logger;
 
     public TrackingHostedService(
         Channel<TrackingMessage> trackingChannel,
-        ITrackingExporterFactory trackingRepositoryFactory)
+        ITrackingExporterFactory trackingRepositoryFactory,
+        ILogger<TrackingHostedService> logger)
     {
         _channelReader = trackingChannel.Reader;
         _trackingRepositoryFactory = trackingRepositoryFactory;
```
[the remark this answers is not in the document]
Comment (gmiserez) [reply src/HotChocolate.Extensions.Tracking/Persistence/TrackingHostedService.cs]: Done.

File: src/HotChocolate.Extensions.Translation/Resources/ResourceClientExtensions.cs [hunk 4]
```diff
@@ -16,7 +15,6 @@ public static class ResourceClientExtensions
                 return res.Value;
             }
 
-            TranslationEventSource.Log.ResourceMissing(key);
```
[the remark this answers is not in the document]
Comment (gmiserez) [reply src/HotChocolate.Extensions.Translation/Resources/ResourceClientExtensions.cs:19]: Implemented the Observer class Feature from enhancement #22 instead. Now a `OnMissingResource` Method of the observer class gets invoked every time a resource is missing. Users can add their own implementation of the `TranslationObserver` to the DI

File: src/HotChocolate.Extensions.Tracking/Persistence/TrackingHostedService.cs [hunk 5]
```diff
@@ -15,30 +15,33 @@ public sealed class TrackingHostedService : BackgroundService
...
+            _logger.LogInformation("Saving tracking message", message.TrackingEntry);
             await _trackingRepositoryFactory.Create(message.TrackingEntry.GetType())
                 .SaveTrackingEntryAsync(message.TrackingEntry, stoppingToken);
         }
     }
 
     public override Task StopAsync(CancellationToken cancellationToken)
     {
-        Log.TrackingPipelineStopped();
+        _logger.LogWarning("Tracking Pipeline Started");
```
[the remark this answers is not in the document]
Comment (gmiserez) [reply src/HotChocolate.Extensions.Tracking/Persistence/TrackingHostedService.cs]: Done.

File: src/HotChocolate.Extensions.Tracking/Persistence/TrackingHostedService.cs [hunk 6]
```diff
@@ -15,30 +15,33 @@ public sealed class TrackingHostedService : BackgroundService
...
+        ITrackingExporterFactory trackingRepositoryFactory,
+        ILogger<TrackingHostedService> logger)
     {
         _channelReader = trackingChannel.Reader;
         _trackingRepositoryFactory = trackingRepositoryFactory;
+        _logger = logger;
     }
 
     protected override async Task ExecuteAsync(CancellationToken stoppingToken)
     {
```
[the remark this answers is not in the document]
Comment (gmiserez) [reply src/HotChocolate.Extensions.Tracking/Persistence/TrackingHostedService.cs:33]: done.

**Review by glucaci — APPROVED** [review]

**Pull request merged**